src/csharp/Microsoft.Spark.Worker.UnitTest/PayloadWriter.cs

```diff
@@ -112,6 +112,44 @@ public void Write(Stream stream, TaskContext taskContext)
         }
     }
 
+    /// <summary>
+    /// TaskContextWriter for version 3.3.*.
+    /// </summary>
+    internal sealed class TaskContextWriterV3_3_X : ITaskContextWriter
+    {
+        public void Write(Stream stream, TaskContext taskContext)
+        {
+            SerDe.Write(stream, taskContext.IsBarrier);
+            SerDe.Write(stream, taskContext.Port);
+            SerDe.Write(stream, taskContext.Secret);
+
+            SerDe.Write(stream, taskContext.StageId);
+            SerDe.Write(stream, taskContext.PartitionId);
+            SerDe.Write(stream, taskContext.AttemptNumber);
+            SerDe.Write(stream, taskContext.AttemptId);
+            SerDe.Write(stream, taskContext.CPUs);
```

Comment:
Please add a comment here to highlight the difference.

Reply:
updated.